### PR TITLE
Expose fullyCompatible parameter of converter

### PR DESCRIPTION
**Pull request by pshao25**
As titled.

### CODE REVIEW

**Review by tadelesh — APPROVED** [review]

File: tools/tsp-client/src/commands.ts [hunk 1]
```diff
@@ -333,6 +333,7 @@ export async function convertCommand(argv: any): Promise<void> {
   const outputDir = argv["output-dir"];
   const swaggerReadme = argv["swagger-readme"];
   const arm = argv["arm"];
+  const fullyCompatible = argv["fullyCompatible"];
```
Comment (catalinaperalta) [inline tools/tsp-client/src/commands.ts:336]: This needs to be using kebab case like all the other flags
```suggestion
  const fullyCompatible = argv["fully-compatible"];
```

File: tools/tsp-client/src/index.ts [hunk 2]
```diff
@@ -199,6 +199,10 @@ const parser = yargs(hideBin(process.argv))
         .option("arm", {
           type: "boolean",
           description: "Convert swagger to ARM TypeSpec",
+        })
+        .option("fullyCompatible", {
```
Comment (catalinaperalta) [inline tools/tsp-client/src/index.ts:203]:
```suggestion
        .option("fully-compatible", {
```

**Review by catalinaperalta — CHANGES_REQUESTED** [review]
Can you please also go into more detail about the flag in the readme under the section for the `convert` command. Otherwise folks may not know what a "fully compatible typespec" is

**Comment by pshao25** [comment]
> Can you please also go into more detail about the flag in the readme under the section for the `convert` command. Otherwise folks may not know what a "fully compatible typespec" is

Updated. I actually planned to update this to TypeSpec-Azure document.

File: tools/tsp-client/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Release
 
+## 2024-10-25 - 0.13.3
+
+- Expose `fullyCompatible` argument for `@autorest/openapi-to-typespec`
```
Comment (catalinaperalta) [inline tools/tsp-client/CHANGELOG.md:5]: Nit: 
```suggestion
- Expose `fully-compatible` argument for `@autorest/openapi-to-typespec`
```

File: tools/tsp-client/README.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ Generate a client library from a TypeSpec project. The `generate` command should
 
 ### convert
 
-Convert an existing swagger specification to a TypeSpec project. This command should only be run once to get started working on a TypeSpec project. TypeSpec projects will need to be optimized manually and fully reviewed after conversion. When using this command a path or url to a swagger README file is required through the `--swagger-readme` flag.
+Convert an existing swagger specification to a TypeSpec project. This command should only be run once to get started working on a TypeSpec project. TypeSpec projects will need to be optimized manually and fully reviewed after conversion. When using this command a path or url to a swagger README file is required through the `--swagger-readme` flag. Highly suggested to leverage TypeSpec built-in libraries with standarded patterns and templates, which will cause discrepancy between the generated TypeSpec and original swagger. If you really don't want this intended discrepancy, add `--fully-compatible` flag.
```
Comment (catalinaperalta) [inline tools/tsp-client/README.md:55]: Nit: 
```suggestion
Convert an existing swagger specification to a TypeSpec project. This command should only be run once to get started working on a TypeSpec project. TypeSpec projects will need to be optimized manually and fully reviewed after conversion. When using this command a path or url to a swagger README file is required through the `--swagger-readme` flag. By default, the converted TypeSpec project will leverage TypeSpec built-in libraries with standard patterns and templates (highly recommended), which will cause discrepancies between the generated TypeSpec and original swagger. If you really don't want this intended discrepancy, add `--fully-compatible` flag to generate a TypeSpec project that is fully compatible with the swagger.
```

**Review by catalinaperalta — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**